proxy/src/bin/proxy.rs

```diff
@@ -632,6 +637,23 @@ fn build_config(args: &ProxyCliArgs) -> anyhow::Result<&'static ProxyConfig> {
         console_redirect_confirmation_timeout: args.webauth_confirmation_timeout,
     };
 
+    let root_store = load_certs()
+        .context("loading native tls certificates")?
+        .clone();
+
+    let client_config =
```

Comment:
Instead of putting more code into the binaries what do you think of a `tls` module where we keep cert loading and client/server config creation and expose functions tailored to external, internal, intra-container and test connections?

Reply:
Done. I also moved some of the existing tls setup to that module.